dask_cuda/benchmarks/local_cupy.py

```diff
@@ -28,75 +29,121 @@ async def _run(client, args):
         await wait(x)
+        end_range(rng)
+
         func_args = (x,)
 
         func = lambda x: (x + x.T).sum()
     elif args.operation == "dot":
+        rng = start_range(message="make array(s)", color="green")
         x = rs.random((args.size, args.size), chunks=args.chunk_size).persist()
         y = rs.random((args.size, args.size), chunks=args.chunk_size).persist()
         await wait(x)
         await wait(y)
+        end_range(rng)
 
         func_args = (x, y)
 
         func = lambda x, y: x.dot(y)
     elif args.operation == "svd":
+        rng = start_range(message="make array(s)", color="green")
         x = rs.random(
             (args.size, args.second_size),
             chunks=(int(args.chunk_size), args.second_size),
         ).persist()
         await wait(x)
+        end_range(rng)
 
         func_args = (x,)
 
         func = lambda x: np.linalg.svd(x)
     elif args.operation == "fft":
+        rng = start_range(message="make array(s)", color="green")
         x = rs.random(
             (args.size, args.size), chunks=(args.size, args.chunk_size)
         ).persist()
         await wait(x)
+        end_range(rng)
 
         func_args = (x,)
 
         func = lambda x: np.fft.fft(x, axis=0)
     elif args.operation == "sum":
+        rng = start_range(message="make array(s)", color="green")
         x = rs.random((args.size, args.size), chunks=args.chunk_size).persist()
         await wait(x)
+        end_range(rng)
+
         func_args = (x,)
 
         func = lambda x: x.sum()
     elif args.operation == "mean":
+        rng = start_range(message="make array(s)", color="green")
         x = rs.random((args.size, args.size), chunks=args.chunk_size).persist()
         await wait(x)
+        end_range(rng)
+
         func_args = (x,)
 
         func = lambda x: x.mean()
     elif args.operation == "slice":
+        rng = start_range(message="make array(s)", color="green")
         x = rs.random((args.size, args.size), chunks=args.chunk_size).persist()
         await wait(x)
+        end_range(rng)
+
         func_args = (x,)
 
         func = lambda x: x[::3].copy()
+    elif args.operation == "col_sum":
+        rng = start_range(message="make array(s)", color="green")
+        x = rs.normal(10, 1, (args.size,), chunks=args.chunk_size).persist()
+        y = rs.normal(10, 1, (args.size,), chunks=args.chunk_size).persist()
+        await wait(x)
+        await wait(y)
+        end_range(rng)
+
+        func_args = (x, y)
+
+        func = lambda x, y: x + y
+    elif args.operation == "col_gather":
+        rng = start_range(message="make array(s)", color="green")
+        x = rs.normal(10, 1, (args.size,), chunks=args.chunk_size).persist()
+        idx = rs.randint(
+            0, len(x), (args.second_size,), chunks=args.chunk_size
+        ).persist()
+        await wait(x)
+        await wait(idx)
+        end_range(rng)
+
+        func_args = (x, idx)
+
+        func = lambda x, idx: x[idx]
```

Comment:
Should we be using `.copy()` here like we do with the slicing operation?

Reply:
Good point, I remember when I wrote CuPy benchmarks, I had to [add `.copy` to ensure it actually slice the array instead of returning a view](https://github.com/pentschev/pybench/blob/89d65a6c418a1fee39d447bd11b8a999835b74a9/pybench/benchmarks/benchmark_array.py#L57). I'm not totally sure whether there's a case where Dask would return a view only, do you know @jakirkham ?

Reply:
That depends on what computational backend Dask is using. If we are using the threaded scheduler, it is probably a view. If we are using the Distributed Scheduler, it may be a view if the data was already on that worker. Otherwise it wouldn't be

Reply:
I guess the safest here is to actually profile both cases. Ideally what we would see if both cases aren't returning a view is:

1. With `.copy()`: some kernels with some copy (or copies) at the end;
2. Without `.copy()`: same kernels as above but not copies at the end.